src/Ocelot.Provider.Consul/ConsulServiceBuilder.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Ocelot.Infrastructure.Extensions;
+using Ocelot.Logging;
+using Ocelot.Provider.Consul.Interfaces;
+using Ocelot.Values;
+
+namespace Ocelot.Provider.Consul;
+
+public class ConsulServiceBuilder : IConsulServiceBuilder
```

Comment:
Why not `DefaultConsulServiceBuilder` ? Then we could propose, extra `LongPollingConsulServiceBuilder`...

Reply:
> Why not `DefaultConsulServiceBuilder`?

Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/2067/commits/da741b4931d881a9224a6ce2936e157ab961a150 ✔️

Reply:
> Then we could propose, extra `LongPollingConsulServiceBuilder`...

I'm sorry, I don't understand. It seems that further refactoring might be beyond the scope of this PR for the #954 bug fix.
You can undertake more extensive refactoring of the Consul provider in subsequent PRs.